src/OpenTelemetry/Trace/Span.cs

```diff
@@ -204,22 +213,12 @@ public void UpdateName(string name)
                 return;
             }
 
-            this.Name = name ?? throw new ArgumentNullException(nameof(name));
+            this.Name = name ?? string.Empty;
         }
 
         /// <inheritdoc/>
         public void SetAttribute(KeyValuePair<string, object> keyValuePair)
         {
-            if (keyValuePair.Key == null)
```

Comment:
Emit logs when failing to add attribute.

Reply:
we are not failing to add it, we add it anyway, but with empty value